module_utils/network_lsr/argument_validator.py

```diff
@@ -271,7 +271,9 @@ def __init__(  # pylint: disable=too-many-arguments
     def _validate_impl(self, value, name):
         v = None
         try:
-            if isinstance(value, self.numeric_type):
+            if isinstance(value, bool):
+                pass
+            elif isinstance(value, self.numeric_type):
                 v = value
             else:
```

Comment:
Seems cleaner to me to avoid pass and make this

`elif not isinstance(value, bool)`

@thom311 what other casts ccould happen here? I guess from other numeric ones to the one that is used here and from string types to the numeric one are expected. Dicts/lists/tuples raise an exception with int at least.

Reply:
`if isinstance(True, int)` is evaluated to `True`.

Reply:
https://stackoverflow.com/questions/60769919/isinstancefalse-int-returns-true

Reply:
Interesting, thank you. Does it make sense to change `isinstance(value, self.numeric_type)` to `type(value) == self.numeric_type`, then?

Reply:
Yes, I think we could. With the current implementation, we can also reject bool safely. Either way is fine.
```python
            if self.numeric_type is not bool and isinstance(value, bool):
                # bool can (probably) be converted to self.numeric_type,
                # but here we don't want to accept a boolean value.
                pass
            elif isinstance(value, self.numeric_type):
                 v = self.numeric_type(value)
            else:
                 v2 = self.numeric_type(value)
                 if isinstance(value, Util.STRING_TYPE) or v2 == value:
                     v = v2
```